Add Strongback to the project

a2aaron opened the pull request:
**TESTED SUCCESSFULLY ON OUR ROBOT!**

Strongback is a framework that extends the existing WPILib Library. It provides wrapper interfaces and classes around many of WPILib's existing hardware classes (for example, it provides wrappers for Gamepads, TalonSRX motors, Solenoids, and so on). The main feature of this is that, unlike WPILib, these classes are testable. A variety of mock classes is provided which allow for unit testing of our classes. This can't be done with WPILib because it expects that all instances of, say a motor, be actually attached to a robot. Thus, if we want the speed of a motor, it has to come from a real TalonSRX. The Motor class's mock instead allows us to manually set an imaginary speed that we can then check against known good values.

a2aaron committed:
Add Strongback to the project

a2aaron committed:
Add Travis CI yaml file

a2aaron committed:
Add TalonSRX libraries

pokofeeftynine committed:
Get the project to build on the robot by adding the right jars to the classpath.

TODO: Fix the hard coded paths.

porglezomp reviewed (commented):
Please use nicer language in commit message

a2aaron committed:
Fix the classpath to not use hard coded paths.

Note that wpilib's build files ONLY look in the wpilib folder, this means that
any jars not located in this folder will not be compiled. Thus, we must
include json.jar, strongback.jar, and any talon libraries in this folder
(these are not visible in the commit). Additionally, json.jar was removed from
this project as it is not longer needed in this repo.

a2aaron committed:
Remove redundant build targets in strongback.xml

These build targets did nothing but call another target, so they were removed.

a2aaron committed:
Make the build target `download-wpilib` download from a git repository URL.

Why? Well, wpilib requires ALL user library jars to be placed in wpilib/user/java/lib,
which is a folder that this repo does not "own". Additionally, there is no easy
or elegant way to move jars into this directory. The simplest solution I can
think of is copying jars into the file from this repo, but I really don't want
to include too many jar files in this repo. Addionally, the site that was
previously downloading WPILib did not include the cpp or tools folders (which
might be nessecarry in order to deploy, I'm not sure).

Instead, I decided to download the wpilib folder directly from a repo that we
"own" (currently it's set to a repo I just made, but we can switch it over to
the FRC1076 organization). As a bonus, this means that updating WPILib is easy,
you update WPILib on a machine through eclipse, and then you push these changes
to GitHub. Then, anyone can pull these updates without having to go through the
same troubles. It also means that dependencies are easier to share between
projects (and only one person needs to ever download these dependencies).

pokofeeftynine committed:
Add strongback testing library and sources for Strongback and JSON.

a2aaron committed:
Update wpilib download repo to point to FRC1076

a2aaron committed:
Override Strongback's incorrect test classpath.

Override Strongback's incorrect buildpath for test compilation. Why?
Strongback assumes that the correct class path is ${classpath}, which
does have WPILib.jar and NetworkTables.jar. However, this doesn't include
any user libraries (like Strongback itself!). Instead we add the path "classpath.path",
which is the correct path that wpilib uses for compilation and includes user libraries.
(the ${classpath} property is defined in ${wpilib.dir}/java/current/ant/build.properties.)

TestRobot.java has been updated to test these changes.

a2aaron committed:
Add testsrc as a source folder

a2aaron committed:
Update Strongback version from 2017.1.0-RC1 to 2017.1.0

a2aaron committed:
Organize property and build file imports.

This commit fixes possible conflicts due to files being imported more than once, as that
could cause some overriding properties to not override properly.

porglezomp commented on build.xml:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
-<project name="FRC Deployment" default="deploy">
+<project name="FRC Deployment" default="test">
```
👍

porglezomp reviewed (approved):
Looks good

porglezomp committed:
Merge branch 'master' into strongback
